sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java

```diff
@@ -19,69 +19,72 @@
 
-  public abstract String getAnalyzerWorkDirProperty();
+  protected abstract String getAnalyzerWorkDirProperty();
 
-  public abstract String getAnalyzerReportDir();
+  protected abstract String getAnalyzerReportDir();
 
-  public boolean isReportsComingFromMSBuild() {
-    if (reportsComingFromMSBuild == null) {
-      reportsComingFromMSBuild = areProtobufReportsPresent();
-    }
-    return reportsComingFromMSBuild;
-  }
+  /**
+   * Returns path of the directory containing all protobuf files, if:
+   * - Property with it's location is defined
+   * - The directory exists
+   * - The directory contains at least one protobuf
+   */
+  public Optional<Path> protobufReportPath() {
+    Optional<String> analyzerWorkDirPath = configuration.get(getAnalyzerWorkDirProperty());
 
-  private boolean areProtobufReportsPresent() {
-    if (!settings.hasKey(getAnalyzerWorkDirProperty())) {
-      return false;
+    if (!analyzerWorkDirPath.isPresent()) {
+      LOG.warn("Property missing: '" + getAnalyzerWorkDirProperty() + "'. No protobuf files will be loaded.");
+      return Optional.empty();
     }
-    Path analyzerOutputDir = protobufReportPathFromScanner();
+
+    Path analyzerOutputDir = Paths.get(analyzerWorkDirPath.get(), getAnalyzerReportDir());
+    configuration.get(getAnalyzerWorkDirProperty());
 
     if (!analyzerOutputDir.toFile().exists()) {
-      LOG.info("Analyzer working directory does not exist");
-      return false;
+      LOG.warn("Analyzer working directory does not exist: " + analyzerOutputDir.toAbsolutePath() + ". No protobuf files will be loaded.");
+      return Optional.empty();
     }
 
-    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.toAbsolutePath().toString().toLowerCase().endsWith(".pb"))) {
+    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.getFileName().toString().toLowerCase().endsWith(".pb"))) {
       long count = StreamSupport.stream(files.spliterator(), false).count();
+      if (count == 0) {
+        LOG.warn("Analyzer working directory contains no .pb file(s). No protobuf files will be loaded.");
+        return Optional.empty();
+      }
       LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
-      return count != 0;
+
+      return Optional.of(analyzerOutputDir);
     } catch (IOException e) {
-      LOG.warn("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath().toString(), e);
-      return false;
+      throw new IllegalStateException("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath(), e);
     }
   }
 
-  public Path protobufReportPathFromScanner() {
-    String analyzerWorkDirPath = settings.getString(getAnalyzerWorkDirProperty());
-    return Paths.get(analyzerWorkDirPath, getAnalyzerReportDir());
+  public Optional<Path> roslynReportPath() {
+    return configuration.get(getRoslynJsonReportPathProperty()).map(p -> Paths.get(p));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)